tsdb/engine/tsm1/engine.go

```diff
@@ -3140,3 +3098,75 @@ func varRefSliceRemove(a []influxql.VarRef, v string) []influxql.VarRef {
 	}
 	return other
 }
+
+const reindexBatchSize = 10000
+
+func (e *Engine) Reindex() error {
+	keys := make([][]byte, reindexBatchSize)
+	seriesKeys := make([][]byte, reindexBatchSize)
+	names := make([][]byte, reindexBatchSize)
+	tags := make([]models.Tags, reindexBatchSize)
+
+	n := 0
+
+	reindexBatch := func() error {
+		if n == 0 {
+			return nil
+		}
+
+		for i, key := range keys[:n] {
+			seriesKeys[i], _ = SeriesAndFieldFromCompositeKey(key)
+			names[i], tags[i] = models.ParseKeyBytesWithTags(seriesKeys[i], tags[i])
```

Comment:
Noticed while removing the log line below... the `tags[i]` arg to `ParseKeyBytesWithTags` is fishy...